src/transformers/generation_beam_search.py

```diff
@@ -309,6 +309,209 @@ def finalize(
         return decoded
 
 
+class DiverseBeamSearchScorer(BeamScorer):
+    r"""
+    :class:`transformers.BeamScorer` implementing diverse beam search decoding.
+
+    Adapted in part from `Facebook's XLM beam search code
+    <https://github.com/facebookresearch/XLM/blob/9e6f6814d17be4fe5b15f2e6c43eb2b2d76daeb4/src/model/transformer.py#L529>`__.
```

Comment:
Could we maybe replace this link with a fitting one in `https://github.com/ashwinkalyan/dbs`?

Reply:
Added the reference for DBS implementation in doc. I have kept `Facebook's XLM beam search code` reference also since most of the part is same as beam search.